Shape_detection/include/CGAL/Polygon_mesh_processing/region_growing.h

```diff
@@ -0,0 +1,338 @@
+
+  @tparam PolygonMesh a model of `FaceListGraph`
+  @tparam RegionMap a model of `WritablePropertyMap` with `boost::graph_traits<PolygonMesh>::%face_descriptor` as key type and `std::size_t` as value type.
+  @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+
+  @param mesh polygon mesh for region growing.
+  @param region_map property map storing the region index of each face. Values start at `0` up to the value returned minus 1.
+         `std::size_t(-1)` is put for faces with no region assigned (can only append if minimum_region_size > 1).
+  @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+
+  @return the number of regions detected
+
+  @todo check that the bug of faces with all vertices in a patch and the face in another patch is gone
+
+  \cgalNamedParamsBegin
+    \cgalParamNBegin{vertex_point_map}
+      \cgalParamDescription{a property map associating points to the vertices of `pmesh`}
+      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+                     as key type and `%Point_3` as value type}
+      \cgalParamDefault{`boost::get(CGAL::vertex_point, pmesh)`}
+      \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+                      must be available in `PolygonMesh`.}
+    \cgalParamNEnd
+    \cgalParamNBegin{geom_traits}
+      \cgalParamDescription{an instance of a geometric traits class}
+      \cgalParamType{a class model of `Kernel`}
+      \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+      \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+    \cgalParamNEnd
+    \cgalParamNBegin{maximum_distance}
+      \cgalParamDescription{the maximum distance from a point to a plane}
+      \cgalParamType{`GeomTraits::FT`}
+      \cgalParamDefault{1}
+    \cgalParamNEnd
+    \cgalParamNBegin{maximum_angle}
+      \cgalParamDescription{the maximum angle in degrees between
+      the normal of the supporting planes of adjacent faces}
+      \cgalParamType{`GeomTraits::FT`}
+      \cgalParamDefault{25 degrees}
+    \cgalParamNEnd
+    \cgalParamNBegin{cosine_value}
+      \cgalParamDescription{the cos value computed as `cos(maximum_angle * PI / 180)`,
+      this parameter can be used instead of the `maximum_angle`}
+      \cgalParamType{`GeomTraits::FT`}
+      \cgalParamDefault{`cos(25 * PI / 180)`}
+    \cgalParamNEnd
+    \cgalParamNBegin{minimum_region_size}
+      \cgalParamDescription{the minimum number of points a region must have}
+      \cgalParamType{`std::size_t`}
+      \cgalParamDefault{1}
+    \cgalParamNEnd
+  \cgalNamedParamsEnd
+ */
+template<
+typename PolygonMesh,
+typename RegionMap,
+typename CGAL_NP_TEMPLATE_PARAMETERS>
+std::size_t
+region_growing_of_planes_on_faces(
```

Comment:
I don't like the name much: is there a point to make it so explicit that it using region growing and planes. Shouldn't it be more the result that matters, hence something like `detect_planar_regions[_on_polygon_mesh]()`?

Reply:
What if we have different algorithm?